pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -155,3 +171,142 @@ func (r *resourceAllocationScorer) isBestEffortPod(podRequests []int64) bool {
+	}
+
+	deviceClassName, exists := deviceClassMapping[resource]
+	if !exists {
+		logger.V(5).Info("Extended resource not found in device class mapping")
+		return 0, 0
+	}
+
+	deviceClass, err := r.draManager.DeviceClasses().Get(deviceClassName)
+	if err != nil {
+		logger.V(5).Info("Failed to get device class for DRA extended resource scoring", "deviceClass", deviceClassName, "error", err)
+		return 0, 0
+	}
+
+	// Only consider device classes that have ExtendedResourceName set
+	// This ensures we only score DRA resources that are meant to be compatible
+	// with device plugin extended resources
+	if ptr.Deref(deviceClass.Spec.ExtendedResourceName, "") != string(resource) {
+		return 0, 0
+	}
+
+	capacity, allocated, err := r.calculateDRAResourceTotals(ctx, node, deviceClass)
+	if err != nil {
+		logger.V(5).Info("Failed to calculate DRA resource capacity and allocated", "deviceClass", deviceClassName, "error", err)
+		return 0, 0
+	}
+
+	logger.V(7).Info("DRA extended resource calculation", "deviceClass", deviceClassName, "capacity", capacity, "allocated", allocated, "podRequest", podRequest)
+	return capacity, (allocated + podRequest)
+}
+
+// calculateDRAResourceTotals computes the total capacity and total allocated count of devices
+// matching the specified Device Class on the given node. It queries the DRA manager for resource
+// slices and allocated devices, filters devices by class and driver, and returns the counts.
+// Returns an error if resource information cannot be retrieved or if node matching fails.
+//
+// Parameters:
+//
+//	ctx         - context for cancellation and deadlines
+//	node        - the node to evaluate device resources on
+//	deviceClass - the device class to filter devices by
+//
+// Returns:
+//
+//	totalCapacity  - total number of devices matching the device class on the node
+//	totalAllocated - number of devices currently allocated from the matching set
+//	error          - any error encountered during processing
+func (r *resourceAllocationScorer) calculateDRAResourceTotals(ctx context.Context, node *v1.Node, deviceClass *resourceapi.DeviceClass) (int64, int64, error) {
+	resourceSlices, err := r.draManager.ResourceSlices().ListWithDeviceTaintRules()
+	if err != nil {
+		return 0, 0, err
+	}
+
+	allocatedDevices, err := r.draManager.ResourceClaims().ListAllAllocatedDevices()
+	if err != nil {
+		return 0, 0, err
+	}
+
+	var totalCapacity, totalAllocated int64
+	for _, slice := range resourceSlices {
+		// filter out slices from other nodes
+		matches, err := structured.NodeMatches(node, ptr.Deref(slice.Spec.NodeName, ""), ptr.Deref(slice.Spec.AllNodes, false), slice.Spec.NodeSelector)
+		if err != nil {
+			return 0, 0, err
+		}
+		if !matches {
+			continue
+		}
+
+		driver := slice.Spec.Driver
+		pool := slice.Spec.Pool.Name
+		// Count devices that match the device class
+		for _, device := range slice.Spec.Devices {
+			if r.deviceMatchesClass(ctx, device, deviceClass, driver) {
+				totalCapacity++
```

Comment:
If [`slice.spec.perDeviceNodeSelection`](https://github.com/kubernetes/kubernetes/blob/74c336734c0eda74f6e9ae9bc82c7470bec7640f/staging/src/k8s.io/api/resource/v1/types.go#L157-L167) is true, then you need to check for each device whether it is available for the node. This needs another unit test case.

Reply:
Done